src/hotspot/share/gc/g1/g1CodeCacheRemSet.cpp

```diff
@@ -286,11 +219,9 @@ class CleanCallback : public StackObj {
 void G1CodeRootSet::clean(HeapRegion* owner) {
   CleanCallback should_clean(owner);
   if (_table != NULL) {
-    int removed = _table->remove_if(should_clean);
-    assert((size_t)removed <= _length, "impossible");
-    _length -= removed;
+    _table->remove_if(should_clean);
```

Comment:
remove_if is a template function but it's only called here.  Does it need to be a template function?

Reply:
It does not, I will fix this.